ts/src/coinspot.ts

```diff
@@ -357,6 +358,73 @@ export default class coinspot extends Exchange {
         return this.parseTrades (trades, market, since, limit);
     }
 
+    async fetchMyTrades (symbol: string = undefined, since: Int = undefined, limit: Int = undefined, params = {}) {
+        /**
+         * @method
+         * @name coinspot#fetchMyTrades
+         * @description fetch all trades made by the user
+         * @param {string} symbol unified market symbol
+         * @param {int} [since] the earliest time in ms to fetch trades for
+         * @param {int} [limit] the maximum number of trades structures to retrieve
+         * @param {object} [params] extra parameters specific to the bitbank api endpoint
+         * @returns {Trade[]} a list of [trade structures]{@link https://github.com/ccxt/ccxt/wiki/Manual#trade-structure}
+         */
+        await this.loadMarkets ();
+        const request = {};
+        let market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+        }
+        if (since !== undefined) {
+            request['startdate'] = this.yyyymmdd (since);
+        }
+        const response = await this.privatePostRoMyTransactions (this.extend (request, params));
+        //  {
+        //   status: 'ok',
+        //   buyorders: [
+        //     {
+        //       otc: false,
+        //       market: 'ALGO/AUD',
+        //       amount: 386.95197925,
+        //       created: '2022-10-20T09:56:44.502Z',
+        //       audfeeExGst: 1.80018002,
+        //       audGst: 0.180018,
+        //       audtotal: 200
+        //     },
+        //   ],
+        //   sellorders: [
+        //     {
+        //       otc: false,
+        //       market: 'SOLO/ALGO',
+        //       amount: 154.52345614,
+        //       total: 115.78858204658796,
+        //       created: '2022-04-16T09:36:43.698Z',
+        //       audfeeExGst: 1.08995731,
+        //       audGst: 0.10899573,
+        //       audtotal: 118.7
+        //     },
+        //   ]
+        // }
+        const buyTrades = this.safeValue (response, 'buyorders', []);
+        for (let i = 0; i < buyTrades.length; i++) {
+            buyTrades[i]['side'] = 'buy';
+            const audTotal = this.safeNumber (buyTrades[i], 'audtotal');
+            buyTrades[i]['total'] = audTotal;
+            const total = this.safeNumber (buyTrades[i], 'total');
+            const amount = this.safeNumber (buyTrades[i], 'amount');
+            buyTrades[i]['price'] = total / amount;
```

Comment:
First `safeNumber` might return a string depending on the number type, second, all of these manipulations should be done inside `parseTrade` and not here.

To do the math we should use strings with the `Precise` module, you can check other examples